src/wizards/tokenSwapDealWizard/components/daoStageContent/daoStageContent.ts

```diff
@@ -33,4 +75,134 @@ export class DaoStageContent {
+        value: address || item,
+      };
+    });
+
+    return Promise.all(res);
+  }
+
+  private get isValidTreasuryAddress(): boolean {
+    return ethers.utils.isAddress(this.data.treasury_address);
+  }
+
+  async refSelectTreasuryChanged(newInputSelectElement) {
+    if (newInputSelectElement && this.treasuryAddresses) {
+      const options = (await this.resolveAddresses(this.treasuryAddresses))
+        .map( address => ({
+          name: address.name,
+          avatarUrl: "",
+          id: address.value,
+          treasury: []}),
+        );
+      newInputSelectElement.setAttribute("options", JSON.stringify(options));
+      newInputSelectElement.addEventListener(DropdownEvent.SelectionChanged, async (e: IAutoCompleteSelectEvent) =>
+      {
+        this.data.treasury_address = e.detail.id;
+        this.form.revalidateErrors();
+      });
+
+      const currentOption = options.find(option => (option.id === this.data.treasury_address));
+      if (currentOption?.name) newInputSelectElement.setAttribute("value", currentOption.name);
+
+      newInputSelectElement.addEventListener(DropdownEvent.Cleared, () => {
+        this.data.treasury_address = "";
+      });
+    }
+  }
+
+  async attached() {
+    if (!this.isHydrated) await this.hydrateDaosList();
+    this.refSelectDAO.setAttribute("options", this.daoListStr);
+    this.refSelectDAO.setAttribute("value", this.data.name);
+    this.isFromDeepDAO = this.data.deepDAOId && !this.data.deepDAOId.startsWith("custom-dao-");
+
+    this.refSelectDAO.addEventListener(DropdownEvent.SelectionChanged, async (e: IAutoCompleteSelectEvent) =>
+    {
+      if (!e.detail.id.startsWith("custom-dao-")) {
+        this.isLoadingDAO = true;
+        this.isFromDeepDAO = true;
+      }
+
+      this.treasuryAddresses = e.detail.treasury || [];
+      if (this.treasuryAddresses.length === 1) {
+        this.data.treasury_address = await this.resolveENS(this.treasuryAddresses[0]);
+      }
+      this.data.name = e.detail.name;
+      this.data.logoURI = e.detail.avatarUrl === DAO_PLACEHOLDER_AVATAR ? "" : e.detail.avatarUrl;
+      this.data.deepDAOId = e.detail.id;
+      this.isLoadingDAO = false;
+      this.form.revalidateErrors();
+    });
+
+    this.refSelectDAO.addEventListener(DropdownEvent.NewAdded, async (e: IAutoCompleteSelectEvent) =>
+    {
+      this.treasuryAddresses = [];
+      this.data.name = e.detail.name;
+      this.data.treasury_address = "";
+      this.data.logoURI = "";
+      this.data.deepDAOId = e.detail.id;
+      this.isFromDeepDAO = false;
+      this.isLoadingDAO = false;
+    });
+
+    this.refSelectDAO.addEventListener(DropdownEvent.Cleared, () => {
+      this.data.name = "";
+      this.data.treasury_address = "";
+      this.treasuryAddresses = [];
+      this.data.logoURI = "";
+      this.data.deepDAOId = "";
+      this.isFromDeepDAO = false;
+    });
+
+  }
+
+  async hydrateDaosList(): Promise<boolean> {
```

Comment:
What is `boolean` used for?
In the 2 usage places, I don't see the return type being used